libraries/api-extractor-model/src/mixins/ApiItemContainerMixin.ts

```diff
@@ -209,6 +274,173 @@ export function ApiItemContainerMixin<TBaseClass extends IApiItemConstructor>(
       return this[_membersByName]!.get(name) || [];
     }
 
+    public findMembersWithInheritance(): IFindApiItemsResult {
+      const messages: IFindApiItemsMessage[] = [];
+      let maybeIncompleteResult: boolean = false;
+
+      const membersByName: Map<string, ApiItem[]> = new Map();
+      const membersByKind: Map<ApiItemKind, ApiItem[]> = new Map();
+
+      const toVisit: ApiItem[] = [];
+      let next: ApiItem | undefined = this;
+
+      while (next) {
+        const membersToAdd: ApiItem[] = [];
+
+        // For each member, check to see if we've already seen a member with the same name
+        // previously in the inheritance tree. If so, we know we won't inherit it, and thus
+        // do not add it to our `membersToAdd` array.
+        for (const member of next.members) {
+          // We add the to-be-added members to an intermediate array instead of immediately
+          // to the maps themselves to support method overloads with the same name.
+          if (ApiNameMixin.isBaseClassOf(member)) {
+            if (!membersByName.has(member.name)) {
+              membersToAdd.push(member);
+            }
+          } else {
+            if (!membersByKind.has(member.kind)) {
+              membersToAdd.push(member);
+            }
+          }
+        }
+
+        for (const member of membersToAdd) {
+          if (ApiNameMixin.isBaseClassOf(member)) {
+            const members: ApiItem[] = membersByName.get(member.name) || [];
+            members.push(member);
+            membersByName.set(member.name, members);
+          } else {
+            const members: ApiItem[] = membersByKind.get(member.kind) || [];
+            members.push(member);
+            membersByKind.set(member.kind, members);
+          }
+        }
+
+        // Interfaces can extend multiple interfaces, so iterate through all of them.
+        const extendedItems: ApiItem[] = [];
+        let extendsTypes: readonly HeritageType[] | undefined;
+
+        switch (next.kind) {
+          case ApiItemKind.Class: {
+            const apiClass: ApiClass = next as ApiClass;
+            extendsTypes = apiClass.extendsType ? [apiClass.extendsType] : [];
+            break;
+          }
+          case ApiItemKind.Interface: {
+            const apiInterface: ApiInterface = next as ApiInterface;
+            extendsTypes = apiInterface.extendsTypes;
+            break;
+          }
+        }
+
+        if (extendsTypes === undefined) {
+          messages.push({
+            messageId: FindApiItemsMessageId.UnsupportedKind,
+            text: `Item ${next.displayName} is of unsupported kind ${next.kind}.`
```

Comment:
I don't understand this logic.  The condition `extendsTypes === undefined` means that the current thing is not a class or an interface; it is some other container such as `ApiEnum`.  If I call `apiEnum.findMembersWithInheritance()`, I would expect it to return the enum members.  The fact that there aren't any "inherited" members seems unimportant.  Whereas this message seems to imply that there is some kind of problem: `"Item apiEnum is of unsupported kind Enum"`

Reply:
> The fact that there aren't any "inherited" members seems unimportant.

Hm, I suppose I was thinking that if someone calls `apiEnum.findMembersWithInheritance()` the message might be useful in communicating essentially "Hey, just so you know, you asked for inherited members, but ApiEnums do not support inheritance". But I can see your point of view, so I'm happy to add a check at the top of this method that basically just defers to `get members()` if the API item is neither an `ApiClass` or `ApiInterface`.